tests/components/evohome/conftest.py

```diff
@@ -117,18 +119,29 @@ def config() -> dict[str, str]:
     }
 
 
-@patch("evohomeasync.broker.Broker._make_request", block_request)
-@patch("evohomeasync2.broker.Broker._client", block_request)
 async def setup_evohome(
     hass: HomeAssistant,
     test_config: dict[str, str],
     install: str = "default",
-) -> MagicMock:
-    """Set up the evohome integration and return its client.
+) -> AsyncGenerator[MagicMock]:
+    """Mock the evohome integration and return its client.
 
     The class is mocked here to check the client was instantiated with the correct args.
     """
 
+    # set the time zone as for the active evohome location
+    loc_idx: int = test_config.get("location_idx", 0)  # type: ignore[assignment]
+
+    try:
+        locn = user_locations_config_fixture(install)[loc_idx]
+    except IndexError:
+        if loc_idx == 0:
+            raise
+        locn = user_locations_config_fixture(install)[0]
+
+    utc_offset: int = locn["locationInfo"]["timeZone"]["currentOffsetMinutes"]  # type: ignore[assignment, call-overload, index]
+    dt_util.set_default_time_zone(timezone(timedelta(minutes=utc_offset)))
```

Comment:
Why do we need to set the timezone here?

Reply:
The date time strings in the vendor's JSON are (local) time zone naive. 

By switching the default time zone to the same as that used by each evohome fixture, it's much easier to compare the calculated date time strings against those in the fixtures.

Reply:
See: #126679